### ISSUE TITLE
[Proposal] Re-name the nested helper (subexpression) to "html-attributes"

### ISSUE THREAD

**Issue by crodriguez1a** [mmun/ember-component-attributes#2]
When the concept was first proposed, @mmun suggested the following:

```hbs
{{my-component (html-attributes aria-labelledby="label-name")}}
```
I wonder if there is consensus to circle back to that naming? It think `html-attributes` more clearly solidifies that the helper is intended to apply native HTML attributes to a component. Whereas, just using `attributes` could potentially confuse new adopters of Ember. I'm anticipating new adopters might try something like:

```hbs
{{my-component (attributes aria-labelledby="label-name"  my-special-thing="my special value")}}

{{! intending to define a property, which should have been written as follows: }}

{{my-component (html-attributes aria-labelledby="label-name") mySpecialThing="my special value"}}
```

I know it seems odd, but I've seen other frameworks that tend to conflate properties and attributes.

**Comment by mmun** [comment]
```hbs
{{my-component mySpecialThing="my special value" (html-attributes aria-labelledby="label-name")}}
```
Small nit, but your second example is not syntactically valid. Positional arguments must come before named arguments.

**Comment by crodriguez1a** [comment]
@mmun Ah! Fixed.

**Comment by rwjblue** [comment]
I'm not opposed to renaming though I don't have an issue with the current name...

**Comment by buschtoens** [comment]
I'm probably stating the obvious, but the advantage of `attributes` over `html-attributes` is that it's five characters shorter and thus a bit faster and easier to squeeze in a single line. Having said that, the advantage of `html-attributes` is that it's more explicit and can't possibly be confused with "regular data attributes".

I and likely many others (hopefully?) tend to write component invocations in multiple lines anyways.

```hbs
{{my-component posArg1 posArg2
  (html-attributes
    role="button"
    aria-labelledby="label-name"
    title=title
  )
  showStuff=true
  levelOfFanciness="over-9000"
  doYouLikeLlamas="absolutely"
  dogOrCatPerson="dog"
}}
```

**Comment by mmun** [comment]
Let's go with `html-attributes`. I'll leave support for `attributes` around for now, but update the README. We can hash out the winner in a future RFC but I'm leaning towards `html-attributes`.

**Comment by rwjblue** [comment]
Seems good to me.  

Any takers on doing the PR?

**Comment by mmun** [comment]
Closed by #5
